tensorflow/python/ops/gradients_util.py

```diff
@@ -172,9 +172,8 @@ def _DefaultGradYs(grad_ys,
               "Gradients of complex tensors must set grad_ys (y.dtype = %r)" %
               y.dtype)
         new_grad_ys.append(
-            array_ops.fill(
-                array_ops.shape(y),
-                constant_op.constant(1, dtype=y.dtype, name="grad_ys_%d" % i)))
+            array_ops.ones(
+                array_ops.shape(y), dtype=y.dtype, name="grad_ys_%d" % i))
```

Comment:
This could be further simplified, but would require changes to
https://github.com/tensorflow/tensorflow/blob/21f11ebd835bcf35c8946925f1e570dc15a1b6e7/tensorflow/python/ops/default_gradient.py#L52-L57
to accept keyword arguments like `name`. Let me know if you think that's worth doing.

Reply:
Does the array_ops.py ones_like work? https://github.com/tensorflow/tensorflow/blob/85c8b2a817f95a3e979ecd1ed95bff1dc1335cff/tensorflow/python/ops/array_ops.py#L3002

Reply:
I don't think that's possible since judging from [this check](https://github.com/tensorflow/tensorflow/blob/f099387426243a6e37e63ad7082bac2fe403efb1/tensorflow/python/ops/gradients_util.py#L198), `y` can be a `tf.resource` which is not supported by `array_ops.ones_like`.

Reply:
None of these (fill, array_ops.zeros, array_ops.zeros_like) support resource-dtype tensors. It looks like it's calling the same "ones" kernel right now, but having it call the standard API would be useful if we ever decide to have a "OnesLike" kernel like we do for zeros.

Reply:
Not sure if I fully understand your comment. The current code works correctly for resource-dtype tensors since `tf.shape` can handle resource-dtype tensors, so `array_ops.zeros` never receives a resource-dtype tensors itself.

Reply:
Shape works fine, but neither constant_op.constant(1, dtype=dtypes.resource) nor array_ops.ones(shape=..., dtype=dtypes.resource) work. array_ops.ones_like just moves some of the boilerplate to a helper, the functionality is equivalent.